### PR TITLE
Fix incorrect dates shown at ends of time ranges

### PR DESCRIPTION
**Pull request by pluehne**
Because of a mistake, the end date of time ranges as shown in the dashboard was not copied by value but by reference. As an effect, displayed dates would be repeatedly decremented by one second every time users hovered data points. For this reason, displayed dates could vary when hovering them multiple times.

This is easily fixed by copying the affected variable by value.

### CODE REVIEW

**Comment by pluehne** [comment]
@toddocon: I discovered this issue while debugging your bug report in #147. I think this one could be related or even the explanation for #147, but I’m not entirely sure. I manually set my time zone to PST, and all dates look fine to me now.

So if you could try out this patch, this would be much appreciated :smile:.

File: docs/assets/js/charts.js [hunk 1]
```diff
@@ -429,7 +429,7 @@ function createHistoryChart(canvas, actionBar)
 
                     // The date range is of the form [t0, t1), inclusive of t0 but exclusive of t1.
                     // Hence, subtract one second from t1 to obtain the previous date in UTC
-                    let t1 = dateRange[1];
```
Comment (larsxschneider) [inline docs/assets/js/charts.js]: Do you see an easy way to cover that with a test case? E.g. move `tooltipTitleCallback` to a "real" function and then test this function with some demo data?

Comment (pluehne) [reply docs/assets/js/charts.js]: I thought about this as well, but now that you think of unit-testing this as well, I probably should do this tomorrow :slightly_smiling_face:.

**Comment by pluehne** [comment]
@larsxschneider: I added some unit tests, so please review this again :slightly_smiling_face:.

**Pull request merged**